sr_utilities_common/scripts/sr_utilities_common/overrun_experiments.py

```diff
@@ -76,13 +77,17 @@ def listener(self):
             raise ValueError('Unrecognized hand type: {}!'.format(self.hand_type))
         elif self.time < 1:
             raise ValueError('Please specify the experiment duration in seconds with -t 60 for example')
-
+        try:
+            rospy.wait_for_message("/" + self.hand_id + "/debug_etherCAT_data", EthercatDebug, timeout=2)
+        except rospy.exceptions.ROSException:
+            rospy.logerr("Cannot find hand id: %s", self.hand_id)
+            raise
```

Comment:
why do you need this raise here? mayb you want to print the error (e.g. the timeout has passed). for instance:
except rospy.ROSException as e:
           rospy.logerr('{}: Could not find the following hand id: {}:'.format(e,
                                                                              self.hand_id))

Reply:
wait_for_message generates an exception when it times out, but the exception message isn't as clear is at could be. I wanted to catch it, let the user know that it's because a hand id was specified that doesn't exist (via a ros error), and then re-raise the wait_for_message exception so that the node actually quits

Reply:
If I print the error but don't re-raise the exception I think it won't kill the node

Reply:
I see! ok, sounds good then :)